### PR TITLE
add gather_nd/tile/masked_select/where/where_index for kunlun

### PR DESCRIPTION
**Pull request by tangzhiyi11**
### PR types
New features

### PR changes
OPs

### Describe
add gather_nd/tile/masked_select/where/where_index for kunlun
*test=kunlun

### CODE REVIEW

File: paddle/fluid/operators/masked_select_op_xpu.cc [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    int ret = xpu::nonzero_count(dev_ctx.x_context(), mask_data, out_size,
+                                 mask->numel());
+    PADDLE_ENFORCE_EQ(ret, XPU_SUCCESS,
+                      platform::errors::External(
+                          "XPU nonzero_count kernel return wrong value[%d %s]",
+                          ret, XPUAPIErrorMsg[ret]));
+
+    xpu_memcpy(static_cast<void*>(&out_size_cpu),
+               static_cast<const void*>(out_size), sizeof(int32_t),
+               XPU_DEVICE_TO_HOST);
```
Comment (taixiurong) [inline paddle/fluid/operators/masked_select_op_xpu.cc:60]: device -> host, 需要加上xpu_wait

File: paddle/fluid/operators/where_index_op_xpu.cc [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    int ret =
+        xpu::nonzero_count(dev_ctx.x_context(), cond_data, true_num, numel);
+    PADDLE_ENFORCE_EQ(
+        ret, XPU_SUCCESS,
+        platform::errors::External(
+            "XPU nonzero_count kernel return wrong value[%d %s] in WhereIndex",
+            ret, XPUAPIErrorMsg[ret]));
+
+    xpu_memcpy(static_cast<void*>(&true_num_cpu),
+               static_cast<const void*>(true_num), sizeof(int32_t),
```
Comment (taixiurong) [inline paddle/fluid/operators/where_index_op_xpu.cc:53]: device -> host 需要加xpu_wait

File: paddle/fluid/operators/masked_select_op_xpu.cc (hunk 1 above)
Comment (tangzhiyi11) [reply paddle/fluid/operators/masked_select_op_xpu.cc:60]: 已加

File: paddle/fluid/operators/where_index_op_xpu.cc (hunk 2 above)
Comment (tangzhiyi11) [reply paddle/fluid/operators/where_index_op_xpu.cc:53]: 已加

**Review by taixiurong — COMMENTED** [review]
LGTM

**Pull request closed without merging**